### ISSUE TITLE
CSS: horizontal scrolling issue with very long lines

### ISSUE THREAD

**Issue by michaelopdenacker** [bootlin/elixir#276]
Facing an issue with the current CSS. The column with the line numbers ends up in the middle of the text. See the attached screenshot.
![Screenshot from 2024-03-05 16-33-34](https://github.com/bootlin/elixir/assets/5461949/a69ee6f7-eb7e-4c2b-a6c6-6ec162e9b99c)

Reported by @minimaxwell 
Investigated by @tleb

**Comment by michaelopdenacker** [comment]
@tleb suggested to remove this from http/banner.css:

> header.header {
>     position: relative !important;
> }

However, it doesn't work for me, unless I zoom out several times.

**Comment by tleb** [comment]
Good URL to reproduce the issue: https://elixir.bootlin.com/linux/v2.6.20.7/source/arch/arm26/ACKNOWLEDGEMENTS

Cannot reproduce on the official instance now that the line in comment https://github.com/bootlin/elixir/issues/276#issuecomment-1979127696 is removed. Both Firefox and Chromium.

I believe the issue is wider than that. Eg see the sidebar that scroll once a viewport width is passed. A proper solution would be to rearrange the HTML. Currently when we scroll horizontally everything is scrolling. We therefore have to make sticky everything that shouldn't scroll (ie everything but the code block).

A better solution would be to make only the code block overflow and handle scrolling at that level. That way there is no issue with the positions of all other blocks in the page.

**Comment by michaelopdenacker** [comment]
Thanks for your investigations! By the way, I reverted the change you suggested on https://elixir.bootlin.com, as it wasn't working for a few of us.

**Comment by fstachura** [comment]
Another case: https://elixir.bootlin.com/linux/latest/source/Documentation/networking/tls-offload-reorder-bad.svg

This happens, because the main element that contains the sidebar and the code is less wide than the longest line of the code (div with the code overflows). The horizontal scrollbar scrolls the whole document, and when the main element is scrolled out of the viewport, it also takes away the sidebar, because the sidebar is contained within. 

![Screenshot_20240309_115430](https://github.com/bootlin/elixir/assets/55913800/c7bbde3c-23d0-4630-a149-76f39717d01e)
![Screenshot_20240309_115440](https://github.com/bootlin/elixir/assets/55913800/e038f375-b33b-4a4b-8782-4a6f3f7504f7)

A quick solution is to make sure that the main element is as wide as the code, and this can be done by setting the `width` property of the main element to `max-content`. 

> A proper solution would be to rearrange the HTML. Currently when we scroll horizontally everything is scrolling. We therefore have to make sticky everything that shouldn't scroll (ie everything but the code block).

I agree. Is the horizontal scrollbar supposed to be as wide as the viewport?

**Comment by tleb** [comment]
> A quick solution is to make sure that the main element is as wide as the code, and this can be done by setting the `width` property of the main element to `max-content`.

I do not think we need to consider a quick and dirty solution. It has been this way since forever. It would take time to develop and test a half-baked solution when we know what the proper should be.

> [...] Is the horizontal scrollbar supposed to be as wide as the viewport?

I do not believe it makes sense. We are scrolling file content; scrollbar should be reflecting that.

**Comment by michaelopdenacker** [comment]
In the meantime, I merged @fstachura 's solution:
https://github.com/bootlin/elixir/commit/a1bff7a948b0a1059b29b7f10f65c1671728f443
